clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -640,6 +666,34 @@ public PartitionRecords(long fetchOffset, TopicPartition partition, List<Consume
             this.partition = partition;
             this.records = records;
         }
+
+        private boolean isConsumed() {
+            return records == null || records.isEmpty();
+        }
+
+        private void discard() {
+            this.records = null;
+        }
+
+        private List<ConsumerRecord<K, V>> take(int n) {
+            if (n > records.size()) {
```

Comment:
No check for `records == null` here. Is the assumption that calling `take` if `records == null` is simply misusage?